designcompose/src/main/java/com/android/designcompose/DesignView.kt

```diff
@@ -435,6 +496,40 @@ internal fun DesignView(
             )
     }
 
+    // triggeredKeyEvent starts out as null and gets populated if any key events that we are
+    // interested in get triggered. We just need to check if it's not null, find the reaction that
+    // the event corresponds with, and execute it.
+    val (triggeredKeyEvent, setTriggeredKeyEvent) =
```

Comment:
It seems a bit funny to do this via a value that gets set and triggers a re-compose to dispatch the side effect. What if the application calls two key injections fast enough that a re-compose doesn't happen until after the second (and the same node is listening for both)? I think that the first one will get squashed.

Does the design break if the `KeyInjectManager` listener directly calls `InteractionState.dispatch`? (I think somehow that ought to cause a re-compose, too, but I've forgotten how).

Reply:
I simplified this by registering the listener with the dispatch data, and then having `KeyInjectManager` call the `dispatch()` function on all listeners directly. This worked, and I believe it's because `InteractionState` changes state internally, triggering a recompose.